src/ideations/ideations.controller.spec.ts

```diff
@@ -1,20 +1,164 @@
 import { Test, TestingModule } from "@nestjs/testing";
 import { IdeationsController } from "./ideations.controller";
 import { IdeationsService } from "./ideations.service";
+import { CreateIdeationDto } from "./dto/create-ideation.dto";
+import { UpdateIdeationDto } from "./dto/update-ideation.dto";
 
 describe("IdeationsController", () => {
     let controller: IdeationsController;
+    let service: IdeationsService;
+
+    const ideationArr = [
+        {
+            id: 1,
+            title: "Ideation 1",
+            description: "Ideation 1 description",
+            vision: "Ideation 1 vision",
+        },
+    ];
+    const ideationOne = ideationArr[0];
+
+    const ideationVoteArr = [{ id: 1, projectIdeaId: 1 }];
+    const ideationVoteOne = ideationVoteArr[0];
+
+    const mockIdeationsService = {
+        createIdeation: jest.fn().mockResolvedValue(ideationOne),
+        createIdeationVote: jest.fn().mockResolvedValue(ideationVoteOne),
+        getIdeationsByVoyageTeam: jest.fn().mockResolvedValue(ideationArr),
+        updateIdeation: jest.fn().mockResolvedValue(ideationOne),
+        deleteIdeation: jest.fn().mockResolvedValue(true),
+        deleteIdeationVote: jest.fn().mockResolvedValue(true),
+    };
 
     beforeEach(async () => {
         const module: TestingModule = await Test.createTestingModule({
             controllers: [IdeationsController],
-            providers: [IdeationsService],
+            providers: [
+                {
+                    provide: IdeationsService,
+                    useValue: mockIdeationsService,
+                },
+            ],
         }).compile();
 
         controller = module.get<IdeationsController>(IdeationsController);
+        service = module.get<IdeationsService>(IdeationsService);
     });
 
     it("should be defined", () => {
         expect(controller).toBeDefined();
     });
+
+    it("should create an ideation", async () => {
+        const userId = "cc1b7a12-72f6-11ee-b962-0242ac120002";
+        const teamId = 1;
+        const createIdeationDto: CreateIdeationDto = {
+            title: "Ideation 1",
+            description: "Ideation 1 description",
+            vision: "Ideation 1 vision",
+        };
+        const req = {
+            user: {
+                userId: userId,
+            },
+        };
+        const ideation = await controller.createIdeation(
+            req,
+            teamId,
+            createIdeationDto,
+        );
+        console.log(ideation);
```

Comment:
we could remove all these console.log (probably won't bother now since it will dismiss all the approvals again) ...